src/services/dataService.js

```diff
@@ -0,0 +1,80 @@
+import * as axios from 'axios';
+import {
+  API_AUTH_URL,
+  API_BASE_URL,
+  DATA_TYPES,
+  API_SEARCH_LOCATION_DATA_TYPE,
+  API_GET_CURRENT_WEATHER_DATA_TYPE,
+  API_GET_FORECAST_DATA_TYPE
+} from '../constants/constants';
+
+const instance = axios.create({
+  withCredentials: true,
+  baseURL: API_BASE_URL
+});
+
+export const weatherAPI = {
+  token: null,
+  getAuthHeaders() {
+    return {
+      headers: {
+        Authorization: `Bearer ${this.token}`
+      }
+    };
+  },
+  async getToken(user, password) {
+    const body = {
+      user,
+      password
+    };
+    try {
+      const response = await instance.post(API_AUTH_URL, body, {
+        headers: {
+          'Content-Type': 'text/plain'
+        }
+      });
+      this.token = response.data.access_token;
+    } catch (error) {
+      console.log(error);
+      this.token = null;
+    }
+
+    return this.token;
+  },
+
+  async searchLocation(query) {
+    const { url, dataKey } = DATA_TYPES[API_SEARCH_LOCATION_DATA_TYPE];
+    try {
+      const response = await instance.get(`${url}${query}`, this.getAuthHeaders());
```

Comment:
Here the second parameter is a bit misleading. It is NOT "headers". It is config that holds headers among other stuff. So I would refactor:

```
getHeaders() {
  return {
        Authorization: `Bearer ${this.token}`
    };
}
```
and 

```
await (http or axios).get(`${url}${locationId}`, { headers : this.getHeaders() });
```

Reply:
changed